sdk/keyvault/keyvault-admin/src/settingsClientModels.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import type { CommonClientOptions, OperationOptions } from "@azure/core-client";
+import { SUPPORTED_API_VERSIONS } from "./constants";
+
+/**
+ * The optional parameters accepted by the KeyVaultSettingsClient.
+ */
+export interface SettingsClientOptions extends CommonClientOptions {
+  /**
+   * The accepted versions of the Key Vault's service API.
+   */
+  serviceVersion?: SUPPORTED_API_VERSIONS;
+
+  /**
+   * Whether to disable verification that the authentication challenge resource matches the Key Vault or Managed HSM domain.
+   * Defaults to false.
+   */
+  disableChallengeResourceVerification?: boolean;
+}
+
+/**
+ * An interface representing the optional parameters that can be passed to {@link KeyVaultSettingsClient.createOrUpdateSetting}
+ */
+export interface CreateOrUpdateSettingOptions extends OperationOptions {}
+
+/**
+ * An interface representing the optional parameters that can be passed to {@link KeyVaultSettingsClient.getSetting}
+ */
+export interface GetSettingOptions extends OperationOptions {}
+
+/**
+ * An interface representing the optional parameters that can be passed to {@link KeyVaultSettingsClient.getSettings}
+ */
+export interface GetSettingsOptions extends OperationOptions {}
+
+/**
+ * Known values of {@link SettingType} that the service accepts.
+ */
+export enum KnownSettingType {
+  /**
+   * A boolean setting.
+   */
+  Boolean = "boolean",
+}
+
+/**
+ * Defines values for KeyVaultSettingType.
+ * {@link KnownSettingType} can be used interchangeably with KeyVaultSettingType. This enum contains
+ * the known values that the service supports.
+ */
+export type KeyVaultSettingType = string;
+
+/**
+ * A setting with a boolean value.
+ */
+export interface BooleanKeyVaultSetting {
+  /**
+   * The name of the setting.
+   */
+  name: string;
+
+  /**
+   * The value of the setting.
+   */
+  value: boolean;
+
+  /**
+   * The type of the setting.
+   */
+  type: KnownSettingType.Boolean;
+}
+
+/**
+ * A setting with an unknown type.
+ */
+export interface DefaultKeyVaultSetting {
```

Comment:
At least for .NET, we don't want to expose anything other than boolean now. Perhaps one day, but not now. It's just opening up a situation that doesn't exist, and we may want to do it differently for actual string settings.

Reply:
After discussion with others on the JS team, I plan on keeping the discriminated union as I just pushed now. This doesn't expose other types of setting, but forces people to check the type of the setting (e.g. `if(setting.kind === "boolean") { ...`) before using it, which increases safety if/when we do add other setting types in the future.